src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManager.java

```diff
@@ -55,6 +56,9 @@ protected <T> T convert(String value, Class<T> type) {
                 return (T) Float.valueOf(value);
             case "Double":
                 return (T) Double.valueOf(value);
+        }}
+        catch (Exception e){
```

Comment:
Never catch a generic exception. This catch block can be triggered for multiple reasons instead of NPE such as a class cast exception, etc. Rather than catching the null pointer exception, add a validation as I have mentioned below.

Reply:
done, but not sure the calling method how to handle it?